agents-and-function-calling/bedrock-agents/features-examples/11-create-agent-using-CDK-with-knowledge-base-and-action-group/bedrockagent/bedrockagent_stack.py

```diff
@@ -0,0 +1,446 @@
+                                                                                 'vectorField': 'bedrock-knowledge-base-default-vector'
+                                                                             },
+                                                                             'vector_size': 1024
+                                                                             }})
+        knowledge_base.node.add_dependency(index_creator)
+
+        # Create Knowlegebase datasource for provisioned S3 bucket
+        datasource = bedrock.CfnDataSource(scope=self,
+                                            id='AgentKBDataSource',
+                                            name='HotelDataS3Source',
+                                            knowledge_base_id=knowledge_base.attr_knowledge_base_id,
+                                            data_source_configuration={'s3Configuration':
+                                                                           {'bucketArn': s3Bucket.bucket_arn},
+                                                                       'type': 'S3'},
+                                            data_deletion_policy='RETAIN')
+
+
+        # Create Lambda role to acess KB
+        base_lambda_policy = iam.ManagedPolicy.from_aws_managed_policy_name(managed_policy_name='service-role/AWSLambdaBasicExecutionRole')
+        kb_lambda_role.add_to_policy(iam.PolicyStatement(sid='SyncKBPolicy', 
+                                                         effect=iam.Effect.ALLOW,
+                                                         resources=[knowledge_base.attr_knowledge_base_arn],
+                                                         actions=['bedrock:StartIngestionJob']))
+        kb_sync_lambda = lambda_.Function(scope=self,
+                        id='SyncKB',
+                        runtime=lambda_.Runtime.PYTHON_3_12,
+                        code=lambda_.Code.from_asset('lambda/kb_sync'),
+                        handler='lambda_function.handler',
+                        timeout=Duration.seconds(300),
+                        role=kb_lambda_role,
+                        environment={'KNOWLEDGE_BASE_ID':knowledge_base.attr_knowledge_base_id,
+                                     'DATA_SOURCE_ID':datasource.attr_data_source_id}
+                        )
+        s3Bucket.add_event_notification(s3.EventType.OBJECT_CREATED,
+                                      s3n.LambdaDestination(kb_sync_lambda))
+        s3Bucket.add_event_notification(s3.EventType.OBJECT_REMOVED,
+                                      s3n.LambdaDestination(kb_sync_lambda))
+
+        # Add an explicit dependency on the lambda, so that the bucket
+        # deployment is started after the lambda is in place
+    
+        bucket_deployment.node.add_dependency(kb_sync_lambda)
+         
+        # Create the DynamoDB table
+        dynamodbable=dynamodb.Table(self, "restaurant_bookings",
```

Comment:
should be parameter -- config file

Reply:
Included dynamodb table name and partition id in config file.